File: devicesetup/src/main/java/io/particle/android/sdk/utils/WorkerFragment.java
package io.particle.android.sdk.utils;


import android.app.Activity;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentActivity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * A simple {@link Fragment} subclass used as a marker for "worker" fragments, while bundling in the
 * fundamental behavior that makes them worker fragments.
 */
public class WorkerFragment extends Fragment {

    // produce a standardized, unique tag
    public static String buildFragmentTag(Class<? extends Fragment> fragClass) {
        // return a unique name
        return "FRAG_" + fragClass.getCanonicalName();
    }

    // Syntactic sugar for simply adding a WorkerFragment
    public static void addFragment(FragmentActivity activity, Fragment frag, String tag) {
        activity.getSupportFragmentManager()
                .beginTransaction()
                .add(frag, tag)
                .commit();
    }


    @Override
    public void onAttach(Activity activity) {
        super.onAttach(activity);
        this.setRetainInstance(true);
    }

    @Override
    public final View onCreateView(LayoutInflater inflater, ViewGroup container,
                                   Bundle savedInstanceState) {
        return null;
    }

}

File: devicesetup/src/main/java/io/particle/android/sdk/utils/WorkerFragment.kt
package io.particle.android.sdk.utils


import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentActivity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

/**
 * A simple [Fragment] subclass used as a marker for "worker" fragments, while bundling in the
 * fundamental behavior that makes them worker fragments.
 */
open class WorkerFragment : Fragment() {


    override fun onAttach(context: Context?) {
        super.onAttach(context)
        this.retainInstance = true
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return null
    }

    companion object {

        // produce a standardized, unique tag
        fun buildFragmentTag(fragClass: Class<out Fragment>): String {
            // return a unique name
            return "FRAG_" + fragClass.canonicalName
        }

        // Syntactic sugar for simply adding a WorkerFragment
        fun addFragment(activity: FragmentActivity, frag: Fragment, tag: String) {
            activity.supportFragmentManager
                    .beginTransaction()
                    .add(frag, tag)
                    .commit()
        }

        // Syntactic sugar for simply adding a WorkerFragment
        fun addFragment(parent: Fragment, frag: Fragment, tag: String) {
            parent.childFragmentManager
                    .beginTransaction()
                    .add(frag, tag)
                    .commit()
        }
    }

}
